cmd/cluster-controller/main.go

```diff
@@ -2,112 +2,78 @@ package main
 
 import (
 	"context"
-	"flag"
+	"errors"
 	"os"
 	"os/signal"
 	"time"
 
+	"github.com/spf13/pflag"
+
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	crdexternalversions "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
-	"k8s.io/kubernetes/pkg/genericcontrolplane/clientutils"
 
 	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
 	kcpexternalversions "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
-	"github.com/kcp-dev/kcp/pkg/reconciler/apiresource"
 	"github.com/kcp-dev/kcp/pkg/reconciler/cluster"
 )
 
-const numThreads = 2
 const resyncPeriod = 10 * time.Hour
 
-var (
-	kubeconfigPath  = flag.String("kubeconfig", "", "Path to kubeconfig")
-	syncerImage     = flag.String("syncer_image", "", "Syncer image to install on clusters")
-	pullMode        = flag.Bool("pull_mode", true, "Deploy the syncer in registered physical clusters in POD, and have it sync resources from KCP")
-	pushMode        = flag.Bool("push_mode", false, "If true, run syncer for each cluster from inside cluster controller")
-	autoPublishAPIs = flag.Bool("auto_publish_apis", false, "If true, the APIs imported from physical clusters will be published automatically as CRDs")
-)
+func bindOptions(fs *pflag.FlagSet) *options {
+	o := options{
+		Options: cluster.BindOptions(cluster.DefaultOptions(), fs),
+	}
+	fs.StringVar(&o.kubeconfigPath, "kubeconfig", "", "Path to kubeconfig")
+	return &o
+}
+
+type options struct {
+	// in the all-in-one startup, client credentials already exist; in this
+	// standalone startup, we need to load credentials ourselves
+	kubeconfigPath string
+	*cluster.Options
+}
+
+func (o *options) Validate() error {
+	if o.kubeconfigPath == "" {
+		return errors.New("--kubeconfig is required")
+	}
+	return o.Options.Validate()
+}
 
 func main() {
 	// Setup signal handler for a cleaner shutdown
 	ctx, cancel := signal.NotifyContext(context.Background(), os.Kill, os.Interrupt)
 	defer cancel()
 
-	flag.Parse()
+	fs := pflag.NewFlagSet("cluster-controller", pflag.ExitOnError)
+	options := bindOptions(fs)
+	if err := fs.Parse(fs.Args()[1:]); err != nil {
```

Comment:
```suggestion
	if err := fs.Parse(os.Args[1:]); err != nil {
```

fs.Args() is unset until after we call `fs.Parse()` - have to pass in command line args.